core/src/components/checkbox/checkbox.tsx

```diff
@@ -204,6 +204,14 @@ export class Checkbox implements ComponentInterface {
     this.ionBlur.emit();
   };
 
+  private onClick = (ev: MouseEvent) => {
+    // The modern control syntax renders a label as a parent element
+    // to the checkbox, which means when the label is clicked, the
+    // checkbox receives this event. To prevent a duplicate click event,
+    // we need to stop the event from bubbling.
+    ev.stopPropagation();
```

Comment:
Does `pointer-events: none` on the native element work here too? We fixed a similar bug in toggle a couple years ago: https://github.com/ionic-team/ionic-framework/pull/23146

I'd prefer a CSS-only approach rather than adding even more JS.

Reply:
`pointer-events: none` on the inner `input` control continues to result in duplicate clicks. The problem is the label invokes a click event on the `input` and does not take into consideration the pointer events.